File: app/src/test/java/com/mysticwind/linenotificationsupport/conversationstarter/model/KeywordEntryTest.java
package com.mysticwind.linenotificationsupport.conversationstarter.model;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class KeywordEntryTest {

    @Test
    public void blankKeywordBecomesEmptyOptional() {
        KeywordEntry entry = KeywordEntry.builder()
                .chatId("chat-1")
                .chatName("Chat 1")
                .keyword(" ")
                .hasReplyAction(true)
                .build();

        assertFalse(entry.getKeyword().isPresent());
        assertTrue(entry.isHasReplyAction());
    }

    @Test
    public void nonBlankKeywordIsPreserved() {
        KeywordEntry entry = KeywordEntry.builder()
                .chatId("chat-1")
                .chatName("Chat 1")
                .keyword("hello")
                .build();

        assertEquals("hello", entry.getKeyword().get());
    }
}

File: app/src/test/java/com/mysticwind/linenotificationsupport/conversationstarter/model/KeywordEntryTest.kt
package com.mysticwind.linenotificationsupport.conversationstarter.model

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class KeywordEntryTest {

    @Test
    fun blankKeywordBecomesEmptyOptional() {
        val entry = KeywordEntry.builder()
            .chatId("chat-1")
            .chatName("Chat 1")
            .keyword(" ")
            .hasReplyAction(true)
            .build()

        assertFalse(entry.keyword.isPresent)
        assertTrue(entry.hasReplyAction)
    }

    @Test
    fun nonBlankKeywordIsPreserved() {
        val entry = KeywordEntry.builder()
            .chatId("chat-1")
            .chatName("Chat 1")
            .keyword("hello")
            .build()

        assertEquals("hello", entry.keyword.get())
    }
}
